### PR TITLE
Mirror Coat stalwart doubles target fix

### PR DESCRIPTION
**Pull request by pacmanboss256**
I don't know what to say here....

### CODE REVIEW

**Comment by pacmanboss256** [comment]
Apparently the `onModifyMove` function in the stalwart and propeller tail ability code did something that messed up the target. deleting the functions there fixes it.

**Comment by DaWoblefet** [comment]
lol nice

**Comment by DaWoblefet** [comment]
Alright @pacmanboss256, it looks like removing ``move.tracksTarget = true;`` causes Stalwart to break with Follow Me and Storm Drain.

**Comment by pacmanboss256** [comment]
that line said it was useless and commented out, so i wonder if i can find a fix for that

**Comment by DaWoblefet** [comment]
Given that it's passing with Ally Switch and the switch-out switch-back-in test, my guess is that the redirection check for Storm Drain, Follow Me, etc. is overriding something earlier than Stalwart can jump in. When you look it over, remember that Dragon Darts is "smartTarget" and Stalwart etc. is "tracksTarget".

**Comment by DaWoblefet** [comment]
I _really_ don't like hardcoding Counter and Mirror Coat here. You're going to encounter the same issue with Metal Burst, probably Magic Coat (idk what that even is supposed to do), and just hardcoding in certain moves doesn't seem like the solution. I would guess that modifying something in ``getTarget()`` in battle.ts, ``useMove()`` in script.ts (particular around the time the ``ModifyMove`` event is ran) or potentially both.

If we did want to use the hardcode solution, at least make sure you hardcode it in both places.

**Comment by pacmanboss256** [comment]
i think in this case something with stalwart using the move modified the target AFTER counter/mirrorcoat modified the target to the correct one, which set target to undefined (?) and ended up random, since the target redirect is before the move is used and modified. 
metalburst, counter and mirror coat already track the target to be the last one that dealt damage, so the target isn't selected in the first place.

**Comment by DaWoblefet** [comment]
Hmm, I can get behind this, since it's no longer hardcoding the moves themselves.

File: data/abilities.ts [hunk 1]
```diff
@@ -2767,11 +2767,7 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 232,
 	},
 	propellertail: {
-		onModifyMove(move) {
```
Comment (DaWoblefet) [inline data/abilities.ts:2770]: You need to make sure the new Stalwart code is in Propeller Tail too.

Comment (pacmanboss256) [reply data/abilities.ts:2770]: oh right durrr
my bad

File: data/abilities.ts [hunk 2]
```diff
@@ -3514,11 +3510,15 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 100,
 	},
 	stalwart: {
-		onModifyMove(move) {
-			// this doesn't actually do anything because ModifyMove happens after the tracksTarget check
-			// the actual implementation is in Battle#getTarget
-			move.tracksTarget = true;
+		onModifyMovePriority: 1,
+		onModifyMove(move, pokemon, target) {
+			// this works with every test....
```
Comment (DaWoblefet) [inline data/abilities.ts:3515]: I'd remove this comment, it doesn't add anything

File: data/abilities.ts [hunk 3]
```diff
@@ -3514,11 +3510,15 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
+			if (!(move.target === 'scripted')) {
+				move.tracksTarget = true;
+			}
 		},
+		// the actual implementation is in Battle#getTarget
+		// move.tracksTarget is not actually useless
```
Comment (DaWoblefet) [inline data/abilities.ts:3521]: I'd remove this comment, it doesn't add anything

File: data/abilities.ts [hunk 4]
```diff
@@ -3514,11 +3517,14 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
-			// this doesn't actually do anything because ModifyMove happens after the tracksTarget check
-			// the actual implementation is in Battle#getTarget
-			move.tracksTarget = true;
+		onModifyMovePriority: 1,
+		onModifyMove(move, pokemon, target) {
+			if (!(move.target === 'scripted')) {
+				move.tracksTarget = true;
+			}
 		},
+		// the actual implementation is in Battle#getTarget
```
Comment (DaWoblefet) [inline data/abilities.ts:3526]: I'd change this to something like "most of the implementation is done in Battle#getTarget". Move the comment inside of onModifyMove (above the if statement).

**Comment by DaWoblefet** [comment]
Make sure the comments in both Stalwart and Propeller Tail are consistent.

**Review by DaWoblefet — APPROVED** [review]

File: data/abilities.ts [hunk 5]
```diff
@@ -2767,10 +2767,12 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
-		onModifyMove(move) {
-			// this doesn't actually do anything because ModifyMove happens after the tracksTarget check
-			// the actual implementation is in Battle#getTarget
-			move.tracksTarget = true;
+		onModifyMovePriority: 1,
+		onModifyMove(move, pokemon, target) {
+			// most of the implementation is in Battle#getTarget
+			if (!(move.target === 'scripted')) {
+				move.tracksTarget = true;
+			}
```
Comment (Marty-D) [inline data/abilities.ts:2775]:
```suggestion
			move.tracksTarget = move.target !== 'scripted';
```

File: data/abilities.ts [hunk 6]
```diff
@@ -3514,10 +3516,12 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
...
```
Comment (Marty-D) [inline data/abilities.ts:3524]:
```suggestion
			move.tracksTarget = move.target !== 'scripted';
```

File: data/abilities.ts [hunk 7]
```diff
@@ -2767,10 +2767,10 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 232,
 	},
 	propellertail: {
-		onModifyMove(move) {
-			// this doesn't actually do anything because ModifyMove happens after the tracksTarget check
-			// the actual implementation is in Battle#getTarget
-			move.tracksTarget = true;
+		onModifyMovePriority: 1,
+		onModifyMove(move, pokemon, target) {
```
Comment (Marty-D) [inline data/abilities.ts:2771]:
```suggestion
		onModifyMove(move) {
```

File: data/abilities.ts [hunk 8]
```diff
@@ -3514,10 +3514,10 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 100,
 	},
 	stalwart: {
-		onModifyMove(move) {
-			// this doesn't actually do anything because ModifyMove happens after the tracksTarget check
-			// the actual implementation is in Battle#getTarget
-			move.tracksTarget = true;
+		onModifyMovePriority: 1,
+		onModifyMove(move, pokemon, target) {
```
Comment (Marty-D) [inline data/abilities.ts:3518]:
```suggestion
		onModifyMove(move) {
```

**Review by Marty-D — APPROVED** [review]
👍

**Pull request merged**